static_frame/core/index_level.py

```diff
@@ -372,6 +372,37 @@ def get_labels(index: Index,
                 next_depth = depth + 1
                 levels.extend((lvl, next_depth) for lvl in level.targets)
 
+    def relabel_at_specific_depth(self,
+            mapper: RelabelInput, depth_level: int
+            ) -> 'IndexLevel':
```

Comment:
Maybe rename to "single" depth?

```suggestion
    def relabel_at_single_depth(self,
            mapper: RelabelInput, 
            depth_level: int,
            ) -> 'IndexLevel':
```